File: scripts/generate-og-image.py
#!/usr/bin/env python3
"""
Generate og.jpg image (1200x630) with dark green background and logo text
"""

try:
    from PIL import Image, ImageDraw, ImageFont
    has_pil = True
except ImportError:
    has_pil = False
    print("PIL/Pillow not installed. Install with: pip install Pillow")
    exit(1)

# Create image
width, height = 1200, 630
img = Image.new('RGB', (width, height), color='#0f4f30')
draw = ImageDraw.Draw(img)

# Try to use a nice font, fallback to default
try:
    # Try to use system font (works on Windows/Mac/Linux)
    font_size = 72
    try:
        font = ImageFont.truetype("arial.ttf", font_size)
    except:
        try:
            font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", font_size)
        except:
            font = ImageFont.load_default()
except:
    font = ImageFont.load_default()

# Draw text "DushanbeMotion" centered
text = "DushanbeMotion"
text_bbox = draw.textbbox((0, 0), text, font=font)
text_width = text_bbox[2] - text_bbox[0]
text_height = text_bbox[3] - text_bbox[1]
text_x = (width - text_width) // 2
text_y = (height - text_height) // 2

draw.text((text_x, text_y), text, fill='#ffffff', font=font)

# Save as JPG
output_path = '../public/og.jpg'
img.save(output_path, 'JPEG', quality=90)
print(f"Created: {output_path}")

File: scripts/generate-og-image.js
// Script to generate og.jpg (1200x630) from logo
// Requires: npm install sharp (or use canvas if available)

const fs = require('fs');
const path = require('path');

// Simple approach: create SVG first, then convert
// For now, we'll create a placeholder that user can replace
// or install sharp to auto-convert

const ogSvg = `<?xml version="1.0" encoding="UTF-8"?>
<svg width="1200" height="630" xmlns="http://www.w3.org/2000/svg">
  <!-- Dark green background -->
  <rect width="1200" height="630" fill="#0f4f30"/>
  
  <!-- Logo/Text centered -->
  <g transform="translate(600, 315)">
    <!-- Use logo SVG if available, otherwise text -->
    <text x="0" y="0" 
          font-family="Arial, sans-serif" 
          font-size="72" 
          font-weight="bold" 
          fill="#ffffff" 
          text-anchor="middle" 
          dominant-baseline="middle">
      DushanbeMotion
    </text>
  </g>
</svg>`;

const outputPath = path.join(__dirname, '../public/og.jpg');

console.log('OG image generation script');
console.log('Note: This script creates SVG. To convert to JPG:');
console.log('1. Install sharp: npm install sharp');
console.log('2. Or use online converter: https://cloudconvert.com/svg-to-jpg');
console.log('3. Or use ImageMagick: convert og.svg og.jpg');

// For now, save as SVG (user will convert manually or we add sharp later)
const svgPath = path.join(__dirname, '../public/og.svg');
fs.writeFileSync(svgPath, ogSvg);
console.log(`Created: ${svgPath}`);

// If sharp is available, convert to JPG
try {
  const sharp = require('sharp');
  sharp(svgPath)
    .resize(1200, 630)
    .jpeg({ quality: 90 })
    .toFile(outputPath)
    .then(() => {
      console.log(`Created: ${outputPath}`);
      fs.unlinkSync(svgPath); // Remove SVG
    })
    .catch(err => {
      console.error('Error converting to JPG:', err.message);
      console.log('SVG saved, please convert manually');
    });
} catch (e) {
  console.log('Sharp not available, SVG saved. Install sharp to auto-convert.');
}

